src/fal/cli/model_generator/module_check.py

```diff
@@ -26,30 +26,52 @@ def generate_dbt_dependencies(module: ast.Module) -> str:
     return "\n".join(map(lambda s: "{{ " + s.strip() + " }}", lines))
 
 
-def _find_function_calls(module: ast.Module) -> List[ast.Call]:
-    return [node for node in ast.walk(module) if isinstance(node, ast.Call)]
+def write_to_model_check(module: ast.Module):
+    """
+    Make sure a there is a single write_to_model function call on the top level
+    """
+
+    all_function_calls = _find_function_calls(ast.walk(module))
+    all_wtm_calls = _filter_function_calls_by_name(all_function_calls, "write_to_model")
+
+    top_level_function_calls = _find_function_calls(
+        stmt.value for stmt in module.body if isinstance(stmt, ast.Expr)
+    )
+    assert (
+        len(all_wtm_calls) == 1 and all_wtm_calls[0] in top_level_function_calls
+    ), "`write_to_model` must be called once on the top level of the Python Model script"
+
```

Comment:
Just to catch up: can the user have multiple `write_to_model` calls (when they are not top-level)? And if so, why are we forbidding the number of top-level `write_to_model` calls?

Reply:
They cannot have non-top-level write_to_models. They can just have a single `write_to_model` and it has to be top level. Maybe we can improve the assert message.

Reply:
seems like we can improve on this, should we create a ticket?